test/test_core_misc.py

```diff
@@ -261,6 +261,67 @@ def test_init_fit_transform(self):
         _ = trained.transform(test_X)
 
 
+class TestTee(unittest.TestCase):
```

Comment:
Would be nice to also demonstrate that Tee can be used for logging, e.g., a few rows of a dataset? Here is an example for how to do that: https://github.com/IBM/lale/blob/master/test/test_core_misc.py#L222

Reply:
It looks like that code is setting up an IO handler, printing, and then getting the values printed.  That seems complex; it is not clear the benefit here -- the listener can, of course, print (that is likely what it will do in practice), but that does not seem critical to test